test/functions_linearalgebra.jl

```diff
@@ -0,0 +1,137 @@
+using LinearAlgebra
+using NamedDims
+using NamedDims: dimnames
+using Test
+
+function baseline_tests(fact, identity)
```

Comment:
do we have any tests for the `iterate` methods defined in this PR?

Reply:
Not explicitly, but there's a `destructing via iteration` test on line 22. `...` is calling iterate under-the-hood which is what prompted me to implement the `iterate` API for these types. If we think that the behaviour of `...` might change in the future I could add an explicit `iterate` test?

Reply:
Nah, that's fine. So long as the iterate methods are being hit in tests :)